### DISCUSSION TITLE
Why is Base URL deprecated?

### DISCUSSION THREAD

**Discussion by vdw-jryn** [grafana/grafana-infinity-datasource#583] [Q&A]
Hello,

I searched for a discussion about this but could not find it. 

I was wondering if someone could explain why the base URL option is deprecated. We currently have a REST-API data source that we use in many different dashboards. To us it seems natural to define one infinity data source as client to that REST-API with the URL already filled in. 

That way:
1. If the URL changes we only need to change it in one location.
2. The chances of anyone making an error by mistyping the URL is reduced.
3. It is easy to see if that data source is used in a dashboard, so we can track the dashboard dependencies. (This can also be done by searching the specific URL in the json of the dashboards, but again if the URL changes we need to change the search parameter while a data source uid should not change).

This is also similar to the way SQL like data sources are set-up within Grafana where you define the database with URL, port, dbname, user,... once and use it in different dashboards.

If someone could explain an error in our reasoning. Or suggest a way to not have to repeat the URL every time we use it. We would appreciate it very much.

Thank you.

**Comment by yesoreyeram** [comment]
Thank you for the feedback. Duplicate of #367. Please update the feedback there for tracking.
